### PR TITLE
[WFLY-12319] exclude not prod artifacts from server's dep management

### PR DESCRIPTION
**Pull request by emmartins**
Issue: https://issues.jboss.org/browse/WFLY-12319

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4967,6 +4977,10 @@
                 <artifactId>wss4j-ws-security-common</artifactId>
                 <version>${version.org.apache.ws.security}</version>
                 <exclusions>
+                    <exclusion>
+                        <groupId>org.apache.geronimo.javamail</groupId>
+                        <artifactId>geronimo-javamail_1.4_mail</artifactId>
+                    </exclusion>
```
Comment (jamezp) [inline pom.xml]: We might want to verify the one below this is valid. It's possible the `groupId` below was just a mistype.

**Review by jamezp — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (emmartins) [reply pom.xml]: verified, dunno if mistyped or simply outdated but no such dep exists there

**Comment by bstansberry** [comment]
@emmartins The CI failure is almost certainly unrelated to this PR, but OTOH it is related to your WFLY-9529 test. So relevant to you :)

https://ci.wildfly.org/viewLog.html?buildId=159274&buildTypeId=WF_PullRequest_Windows

It's the only failure so far of those tests.

**Comment by emmartins** [comment]
@bstansberry looking at server log this happened due to:

`2019-07-25 00:32:46,785 WARN  [org.apache.activemq.artemis.ra] (default-threads - 1) AMQ152005: Failure in broker activation org.apache.activemq.artemis.ra.inflow.ActiveMQActivationSpec(ra=org.wildfly.extension.messaging.activemq.ActiveMQResourceAdapter@798f9ea7 destination=java:app/mdb destinationType=javax.jms.Queue ack=Auto-acknowledge durable=false clientID=null user=null maxSession=15): java.lang.NullPointerException
	at org.apache.activemq.artemis.ra.inflow.ActiveMQActivation.getAddress(ActiveMQActivation.java:518)
	at org.apache.activemq.artemis.ra.inflow.ActiveMQMessageHandler.setup(ActiveMQMessageHandler.java:172)
	at org.apache.activemq.artemis.ra.inflow.ActiveMQActivation.setup(ActiveMQActivation.java:317)
	at org.apache.activemq.artemis.ra.inflow.ActiveMQActivation$SetupActivation.run(ActiveMQActivation.java:731)
	at org.wildfly.extension.messaging.activemq.ActiveMQResourceAdapter$WorkWrapper.run(ActiveMQResourceAdapter.java:171)
	at org.jboss.jca.core.workmanager.WorkWrapper.runWork(WorkWrapper.java:445)
	at org.jboss.as.connector.services.workmanager.WildflyWorkWrapper.runWork(WildflyWorkWrapper.java:69)
	at org.jboss.jca.core.workmanager.WorkWrapper.run(WorkWrapper.java:223)
	at org.jboss.threads.SimpleDirectExecutor.execute(SimpleDirectExecutor.java:29)
	at org.jboss.threads.QueueExecutor.runTask(QueueExecutor.java:789)
	at org.jboss.threads.QueueExecutor.access$100(QueueExecutor.java:44)
	at org.jboss.threads.QueueExecutor$Worker.run(QueueExecutor.java:809)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:485)`

A sec earlier there was this suspicious warning:

`2019-07-25 00:32:45,398 WARN  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 42) AMQ222205: OutOfMemoryError possible! There are currently 62 addresses with a total max-size-bytes of 650,117,120 bytes, but the maximum memory available is 514,850,816 bytes.`

I restarted the job, most probably nothing to worry about with the PR or the failed test, also considering its run history, but maybe @ehsavoie should investigate the server.log as followup, here is a direct link to it:

https://ci.wildfly.org/repository/download/WF_PullRequest_Windows/159274:id/testsuite/integration/basic/target/wildfly/standalone/log/server.log

**Pull request merged**